modules/frontend/searchsharding.go

```diff
@@ -69,11 +71,17 @@ func (r *searchResponse) addResponse(res *tempopb.SearchResponse) {
 	r.mtx.Lock()
 	defer r.mtx.Unlock()
 
-	r.results.Traces = append(r.results.Traces, res.Traces...)
+	for _, t := range res.Traces {
+		// just take the first
+		if _, ok := r.resultsMap[t.TraceID]; !ok {
+			r.resultsMap[t.TraceID] = t
```

Comment:
**suggestion**: This probably needs to dedupe/combine search results, for example the start/end/rootspan may be spread across multiple pages/blocks, and the individual segments that matched don't have the complete picture.  But not critical, can be done later (if we decide it is needed).    Add todo for now?